Tests/Runner/BranchedWorkspaceTest.php

```diff
@@ -0,0 +1,281 @@
+        $storageApi = $this->getClient();
+        $storageApiMaster = $this->getMasterClient();
+
+        // prepare data
+        $inBucketId = $this->cleanupAndCreateBucket($storageApi, self::BUCKET_NAME, $storageApi::STAGE_IN);
+        $outBucketId = $this->cleanupAndCreateBucket($storageApi, self::BUCKET_NAME, $storageApi::STAGE_OUT);
+
+        $this->loadDataIntoTable($storageApi, $inBucketId, 'test-table', [
+            ['id', 'text'],
+            ['test1', 'test1'],
+        ]);
+
+        // create branch
+        $branchesApiMaster = new DevBranches($storageApiMaster);
+        $branchId = $this->cleanupAndCreateBranch($branchesApiMaster, self::BRANCH_NAME);
+        $storageApiWrapper = new ClientWrapper($storageApi, null, null, $branchId);
+
+        // setup configuration outside branch
+        $componentsApiForConfig = new Components($storageApi);
+        $configId = $this->generateConfigId();
+
+        $this->cleanupAndCreateConfiguration(
+            $componentsApiForConfig,
+            (new Configuration())
+                ->setComponentId('keboola.runner-workspace-test')
+                ->setName($configId)
+                ->setConfigurationId($configId)
+        );
+
+        // run testing job
+        $this->expectException(ClientException::class);
+        $this->expectExceptionMessage(sprintf('Configuration %s not found', $configId));
+
+        $this->runCopyJob($storageApiWrapper, $configId, $inBucketId, $outBucketId, self::TABLE_NAME);
+    }
+
+    public function testBranchConfigIsAvailableToBranchJob()
+    {
+        $storageApi = $this->getClient();
+        $storageApiMaster = $this->getMasterClient();
+
+        // prepare data
+        $inBucketId = $this->cleanupAndCreateBucket($storageApi, self::BUCKET_NAME, $storageApi::STAGE_IN);
+        $outBucketId = $this->cleanupAndCreateBucket($storageApi, self::BUCKET_NAME, $storageApi::STAGE_OUT);
+
+        $this->loadDataIntoTable($storageApi, $inBucketId, 'test-table', [
+            ['id', 'text'],
+            ['test1', 'test1'],
+        ]);
+
+        // create branch
+        $branchesApiMaster = new DevBranches($storageApiMaster);
+        $branchId = $this->cleanupAndCreateBranch($branchesApiMaster, self::BRANCH_NAME);
+        $storageApiWrapper = new ClientWrapper($storageApi, null, null, $branchId);
+
+        // setup configuration inside branch
+        $componentsApiForConfig = new Components($storageApiWrapper->getBranchClient());
+        $configId = $this->generateConfigId();
+
+        $this->createConfiguration(
+            $componentsApiForConfig,
+            (new Configuration())
+                ->setComponentId('keboola.runner-workspace-test')
+                ->setName('runner-tests')
+                ->setConfigurationId($configId)
+        );
+
+        // run testing job
+        $this->runCopyJob($storageApiWrapper, $configId, $inBucketId, $outBucketId, self::TABLE_NAME);
+    }
+
+    private function generateConfigId()
```

Comment:
programuješ v testech :)

Reply:
Fakt? Drzel jsem se pravidla "zadny if, zadny for" :) Ne, ted vazne, co je na tom spatne? V ostatnich testech je configId presne takle generovany, jen se v kazdem testu vola primo `uniqid`. Me prislo akorat vhodny to vytahnout do metody, ze pak vsechny tes case generuji to ID zarucene stejnym zpusobem a nemusim nad tim premyslet.